server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -173,7 +173,7 @@ public final EngineConfig config() {
      * Return the latest active SegmentInfos from the engine.
      * @return {@link SegmentInfos}
      */
-    protected abstract SegmentInfos getLatestSegmentInfos();
+    public abstract SegmentInfos getLatestSegmentInfos();
```

Comment:
I don't think we need this additional method to fetch SegmentInfos, we already have `getSegmentInfosSnapshot` on engine.  Engine provides a default implementation, so we'll need to override this in NRTReplicationEngine.